runtime/src/accounts_db.rs

```diff
@@ -759,6 +758,14 @@ impl AccountsDB {
         let hash_info = bank_hashes
             .get(&parent_slot)
             .expect("accounts_db::set_hash::no parent slot");
+        if bank_hashes.get(&slot).is_some() {
```

Comment:
is this additional line in scope for this PR?

Reply:
@rob-solana Yes!

First, this PR added additional tests for the lazy initialization of slot hashes sysvar around [here](https://github.com/solana-labs/solana/pull/7892/files#diff-a7549f152920d85fb44e6a784b8e5e1fR3954). That exact line internally hits  here and bails out with this `if` condition. So, this `if` is needed to pass the test correctly. Otherwise, [this following assertion](https://github.com/solana-labs/solana/pull/7892/files#diff-a7549f152920d85fb44e6a784b8e5e1fR3956) for idempotent bank hash would fail because `set_file` without this `if` would [incorrectly reset the bank_hash](https://github.com/solana-labs/solana/pull/7892/files#diff-2099c5256db4eb5975c8834af38f6456R774) for the tested slot, _which is shared by `bank2` and `bank3`_.

I know such sharing of a slot between child banks doesn't occur in the real validator behavior. Thus, I alternatively could replace this `error!` with `panic!` and adjust all tests. But it takes extra effort and introduces an artifical limitation to AccountsDB: Currently, all of other AccountsDB code doesn't forbid such slot sharing, it just works. Only this part didn't work; So I wanted to preserve that flexible property and managed to do that with this tiny additional `if` clause.

In other words, AccountsDB isn't aware of anything called _forks_ and free of leaky abstraction now and forever. I think that's nice and beautiful design and separation of concern.  :)